xrpl/models/transactions/transaction.py

```diff
@@ -88,6 +88,39 @@ class Signer(BaseModel):
     #: This field is required.
     signing_pub_key: str = REQUIRED  # type: ignore
 
+    @classmethod
+    def is_dict_of_model(cls: Type[Signer], dictionary: Dict[str, Any]) -> bool:
+        """
+        Returns True if the input dictionary was derived by the `to_dict`
+        method of an instance of this class. In other words, True if this is
+        a dictionary representation of an instance of this class.
+
+        NOTE: does not account for model inheritance, IE will only return True
+        if dictionary represents an instance of this class, but not if
+        dictionary represents an instance of a subclass of this class.
+
+        Args:
+            dictionary: The dictionary to check.
+
+        Returns:
+            True if dictionary is a dict representation of an instance of this
+            class.
+        """
+        return (
+            isinstance(dictionary, dict)
+            and "signer" in dictionary
+            and super().is_dict_of_model(dictionary["signer"])
+        )
+
+    def to_dict(self: Signer) -> Dict[str, Any]:
+        """
+        Returns the dictionary representation of a Signer.
+
+        Returns:
+            The dictionary representation of a Signer.
+        """
+        return {"signer": super().to_dict()}
```

Comment:
this object needs to be wrapped in a `"signer"` key

Reply:
This is the convention `rippled` uses whenever it serializes an "Object in an array", so you'll see it in other places too like in [transaction metadata](https://xrpl.org/transaction-metadata.html#example-metadata). It's always:

```json
{
  "ObjectTypeName": {
      "object fields here": true,
      "etc.": "..."
  }
}
```